### DISCUSSION TITLE
Number parsing: is fallback necessary?

### DISCUSSION THREAD

**Discussion by piotrrzysko** [simdjson/simdjson#2048] [Q&A] [answered]
I've been working on implementing float parsing in [simdjson-java](https://github.com/simdjson/simdjson-java), and while researching the topic, I've noticed that simdjson, in some cases, requires [fallback to a more accurate multiplication of multiword integers](https://github.com/simdjson/simdjson/blob/f043db4dcaed08513b9bce798e4ef84a62cdf7f8/include/simdjson/generic/numberparsing.h#L207). According to [Fast Number Parsing Without Fallback](https://arxiv.org/abs/2212.06644), the fallback is not necessary. 

Do I understand correctly that we can remove the fallback? If that's the case, I'd be happy to submit a PR.

**Comment by piotrrzysko** [comment]
@lemire Apologies for pinging you. I would appreciate your input on this topic. In simdjson-java, I've entirely skipped the fallback: https://github.com/simdjson/simdjson-java/blob/567af8f7b2e025cf7ac3c7f243a1ab8ac58b3f35/src/main/java/org/simdjson/NumberParser.java#L172.

**Comment by lemire** [comment, marked as answer]
@piotrrzysko You are correct that the multiplication, after loading 128 bits, is exact. PR invited!
